src/core.jl

```diff
@@ -1370,15 +1370,37 @@ Base.haskey(graph::Graph, name) = isnull(get_node_by_name(graph, name))
 
 
 
+node_name(::Void) = nothing
+node_name(xs::AbstractVector)=node_name.(xs)
```

Comment:
Tested it, doesn't work out.
Because we don't always know when we expect an array.
`gradients(y, x, grad_y)`
can have any of its 3 parameters as an array of Tensors or as a Tensor, (though if `y` is an array `y_grad` must also).

And on 0.5 at least `nodename.(x)` for `x::Tensor` comes back with: 
```julia
  MethodError: no method matching start(::TensorFlow.Tensor{Int32})
```

So with that trialed, I will merge this now, as is.